Library/Homebrew/test/test_utils.rb

```diff
@@ -210,4 +210,17 @@ def test_number_readable
     assert_equal "1,000", number_readable(1_000)
     assert_equal "1,000,000", number_readable(1_000_000)
   end
+
+  def test_truncate_text_to_approximate_size
+    glue = "\n[...snip...]\n" # hard-coded copy from truncate_text_to_approximate_size
+    n = 20
+    long_s = "x" * 40
+    s = truncate_text_to_approximate_size(long_s, n)
+    assert_equal s.length, n
+    assert_match(/x+#{Regexp.escape(glue)}x+/, s)
+    s = truncate_text_to_approximate_size(long_s, n, :front_weight => 0.0)
+    assert_equal s, glue + ("x" * (n-glue.length))
+    s = truncate_text_to_approximate_size(long_s, n, :front_weight => 1.0)
+    assert_equal s, ("x" * (n-glue.length)) + glue
```

Comment:
Same here. The expected value needs to be the first argument to any `assert_*` method, otherwise the output will be confusing once the assertion isn't satisfied.